src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt

```diff
@@ -223,6 +225,15 @@ class AssessmentService(
 
     val savedAssessment = assessmentRepository.save(assessment)
 
+    if (placementRequirements != null) {
+      val placementRequestValidationResult = placementRequestService.createPlacementRequest(assessment, placementRequirements)
+
+      when (placementRequestValidationResult) {
+        is ValidatableActionResult.Success -> Unit
+        else -> return AuthorisableActionResult.Success(placementRequestValidationResult.translateError())
```

Comment:
This threw me a bit, should be we returning `Success` if it's an error?

Reply:
I feel like an `if` would be better here, then we don't have to return a `Unit`

Reply:
> should be we returning Success if it's an error?

This is because the return type for the function is `AuthorisableActionResult<ValidatableActionResult<AssessmentEntity>>` so here we're returning `yes this user is authorised to make this change but there is a validation error with the parameters provided`

Agreed on the `if` point - have updated